ecosystem/model.py

```diff
@@ -86,21 +87,18 @@ def count_type(model, patch_condition):
                 count += 1
         return count
 
-
-    @staticmethod
-    def calculate_local_densities(model):
+    def calculate_local_densities(self, model):
         '''Helper method to count vegetated neighbours.'''
 
         qplusplus = []
-        qplusminus = []
+        qminusplus = []
 
         for patch in model.schedule.agents:
```

Comment:
Maybe this line should say self.schedule.agents instead of model.schedule.agents, since this method is no longer static?